### PR TITLE
further improve pull request template

### PR DESCRIPTION
**Pull request by juarezr**
This PR has the objetive of improving comunication in PR template.
At least for me. :smile:

## Changes

1. Improved the pull request template
2. Split items in categories.
3. Added new entries to signaling PR status

## Checklist

Checklist for for pull requests including new code and/or changes to existing code...

* [x] ~~Code~~
  * [ ] ~~Includes unit tests~~
  * [ ] ~~New functions have docstrings with examples that can be run with doctest~~
  * [ ] ~~New functions are included in API docs~~
  * [ ] ~~Docstrings include notes for any changes to API or behaviour~~
  * [ ] ~~All changes documented in docs/changes.rst~~
* [x] ~~Testing~~
  * [ ] ~~\(Optional) Tested local against remote servers~~
  * [ ] ~~Travis CI passes (unit tests run under Linux)~~
  * [ ] ~~AppVeyor CI passes (unit tests run under Windows)~~
  * [ ] ~~Unit test coverage has not decreased (see Coveralls)~~
* [x] Changes
  * [ ] ~~\(Optional) Just a proof of concept~~
  * [ ] ~~\(Optional) Work in progress~~
  * [x] Ready to review
  * [x] Ready to merge

### Legend

* [ ] Not done yet
* [x] Finished
* [ ] ~~Doesn't apply to this PR~~

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -1,16 +1,36 @@
-<!--
-Description of PR
--->
 
-<!--
+This PR has the objetive of...
```
Comment (alimanfoo) [inline .github/PULL_REQUEST_TEMPLATE.md:2]:
```suggestion
This PR has the objective of...
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -1,16 +1,36 @@
...
+  * [ ] Travis CI passes (unit tests run under Linux)
+  * [ ] AppVeyor CI passes (unit tests run under Windows)
+  * [ ] Unit test coverage has not decreased (see Coveralls)
+* [ ] Changes
+  * [ ] \(Optional) Just a proof of concept
+  * [ ] \(Optional) Work in progress
+  * [ ] Ready to review
+  * [ ] Ready to merge
+
+### Legend
```
Comment (alimanfoo) [inline .github/PULL_REQUEST_TEMPLATE.md:32]: What does "legend" mean?

Comment (juarezr) [reply .github/PULL_REQUEST_TEMPLATE.md]: In portuguese:
  legenda => reference, subtitle, caption, description
Easily mistaken with:
  lenda => legend, myth, tale, fable

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (juarezr) [reply .github/PULL_REQUEST_TEMPLATE.md]: spelling :+1:

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -1,16 +1,36 @@
...
+  * [ ] \(Optional) Just a proof of concept
+  * [ ] \(Optional) Work in progress
+  * [ ] Ready to review
+  * [ ] Ready to merge
+
+### Reference
+
+* [ ] Not done yet
+* [x] Finished
+* [x] ~~Doesn't apply to this PR~~
```
Comment (alimanfoo) [inline .github/PULL_REQUEST_TEMPLATE.md:36]: Hi @juarezr, sorry I didn't get it first time, now I see what is meant by the "Legend" section :-)

FWIW this section was a bit confusing to me, I thought these were more check-boxes the PR submitter was supposed to complete. Hopefully most PR submitters will understand how to use checkboxes and do something sensible if a line does not apply to the PR, suggest you could remove this section.

Comment (juarezr) [reply .github/PULL_REQUEST_TEMPLATE.md]: I agree.

**Comment by alimanfoo** [comment]
Thanks @juarezr :smile:

**Pull request merged**